crates/orchestrator/src/tests/config.rs

```diff
@@ -1,49 +1,97 @@
 use std::sync::Arc;
 
-use crate::config::{
-    build_alert_client, build_da_client, build_prover_service, build_settlement_client, config_force_init, Config,
-};
-use crate::data_storage::DataStorage;
-use da_client_interface::DaClient;
+use aws_config::SdkConfig;
 use httpmock::MockServer;
-
-use crate::alerts::Alerts;
-use prover_client_interface::ProverClient;
-use settlement_client_interface::SettlementClient;
 use starknet::providers::jsonrpc::HttpTransport;
 use starknet::providers::{JsonRpcClient, Url};
+
+use da_client_interface::{DaClient, MockDaClient};
+use prover_client_interface::{MockProverClient, ProverClient};
+use settlement_client_interface::{MockSettlementClient, SettlementClient};
 use utils::env_utils::get_env_var_or_panic;
 use utils::settings::default::DefaultSettingsProvider;
 
-use crate::database::mongodb::config::MongoDbConfig;
-use crate::database::mongodb::MongoDb;
-use crate::database::{Database, DatabaseConfig};
-use crate::queue::sqs::SqsQueue;
-use crate::queue::QueueProvider;
-use crate::tests::common::{create_sns_arn, create_sqs_queues, drop_database, get_storage_client};
+use crate::alerts::{Alerts, MockAlerts};
+use crate::config::{
+    build_alert_client, build_da_client, build_database_client, build_prover_service, build_queue_client,
+    build_settlement_client, Config,
+};
+use crate::data_storage::{DataStorage, MockDataStorage};
+use crate::database::{Database, MockDatabase};
+use crate::queue::{MockQueueProvider, QueueProvider};
+use crate::tests::common::{create_sns_arn, create_sqs_queues, drop_database};
+
+use super::common::get_storage_client;
 
 // Inspiration : https://rust-unofficial.github.io/patterns/patterns/creational/builder.html
 // TestConfigBuilder allows to heavily customise the global configs based on the test's requirement.
 // Eg: We want to mock only the da client and leave rest to be as it is, use mock_da_client.
 
+pub enum MockType {
+    StarknetClient(Arc<JsonRpcClient<HttpTransport>>),
+    DaClient(Box<dyn DaClient>),
+    ProverClient(Box<dyn ProverClient>),
+    SettlementClient(Box<dyn SettlementClient>),
+
+    Alerts(Box<dyn Alerts>),
+    Database(Box<dyn Database>),
+    Queue(Box<dyn QueueProvider>),
+    Storage(Box<dyn DataStorage>),
+}
+
+// By default, everything is on Dummy.
+pub enum ConfigType {
```

Comment:
so the goal is that when initializing a service we can chose wether to have a mock, the actual type which is the one used in prod ? and a dummy one, which is never used ?

Reply:
Very precisely, yes.

Reply:
I'm not 100% sure of the approach but can be revisited later anyways